### PR TITLE
Feature/creat backend

### PR DESCRIPTION
**Pull request by ViacheslavKechkin**

### CODE REVIEW

File: src/modules/controllers/task.controller.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const Task = require('../../db/models/task/index');
+
+module.exports.getAllTasks = (req, res, next) => {
+  Task.find().then(result => {
+    res.send({ data: result });
+  });
+};
+
+module.exports.createNewTask = (req, res, next) => {
+  const task = new Task(req.body);
```
Comment (NataliaShchelkunova) [inline src/modules/controllers/task.controller.js:10]: whats happends if i don't send data ?

File: src/modules/controllers/task.controller.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  task.save().then(result => {
+    res.send(result);
+  }).catch(err => console.log(err));
+};
+
+module.exports.changeTaskInfo = (req, res, next) => {
+  Task.updateOne({ _id: req.body._id }, req.body).then(result => {
```
Comment (NataliaShchelkunova) [inline src/modules/controllers/task.controller.js]: whats happends if i don't send id or another data?

Comment (NataliaShchelkunova) [reply src/modules/controllers/task.controller.js]: destructure req.body._id

File: src/modules/controllers/task.controller.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    Task.find({ _id: req.body._id }).then(result => {
+      res.send(result);
+    });
+  });
+};
+
+module.exports.deleteTask = (req, res, next) => {
+  Task.deleteOne({ _id: req.body._id }).then(result => {
```
Comment (NataliaShchelkunova) [inline src/modules/controllers/task.controller.js]: whats happends if i don't send id? Pls send to the client all Tasks without removed task

File: src/modules/controllers/task.controller.js (hunk 1 above)
Comment (ViacheslavKechkin) [inline src/modules/controllers/task.controller.js:10]: A new object will be added without data but with the assigned id

File: src/modules/controllers/task.controller.js (hunk 2 above)
Comment (ViacheslavKechkin) [reply src/modules/controllers/task.controller.js]: err 422

File: src/modules/controllers/task.controller.js (hunk 3 above)
Comment (ViacheslavKechkin) [reply src/modules/controllers/task.controller.js]: The task will not be deleted

File: src/modules/controllers/task.controller.js [hunk 4]
```diff
@@ -14,15 +14,17 @@ module.exports.createNewTask = (req, res, next) => {
 };
 
 module.exports.changeTaskInfo = (req, res, next) => {
-  Task.updateOne({ _id: req.body._id }, req.body).then(result => {
-    Task.find({ _id: req.body._id }).then(result => {
+  const {id} = req.body;
```
Comment (NataliaShchelkunova) [inline src/modules/controllers/task.controller.js]: code style

File: src/modules/controllers/task.controller.js [hunk 5]
```diff
@@ -14,15 +14,17 @@ module.exports.createNewTask = (req, res, next) => {
...
+  Task.updateOne({ _id: id }, id).then(result => {
+    Task.find({ _id: id }).then(result => {
```
Comment (NataliaShchelkunova) [inline src/modules/controllers/task.controller.js]: this will work uncorrectly

File: src/modules/controllers/task.controller.js (hunk 4 above)
Comment (NataliaShchelkunova) [inline src/modules/controllers/task.controller.js]: code style

File: src/modules/controllers/task.controller.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    });
+  }).catch(err => console.log(err));
+};
+
+module.exports.changeTaskInfo = (req, res, next) => {
+  const { id } = req.body;
+  Task.updateOne({ _id: id }, id).then(result => {
+    Task.find({ _id: id }).then(result => {
+      //добавить что то в этом месте
+      console.log(result);
```
Comment (NataliaShchelkunova) [inline src/modules/controllers/task.controller.js]: what is it?

File: src/modules/controllers/task.controller.js (hunk 4 above)
Comment (NataliaShchelkunova) [inline src/modules/controllers/task.controller.js]: code style

File: src/modules/controllers/task.controller.js (hunk 5 above)
Comment (NataliaShchelkunova) [inline src/modules/controllers/task.controller.js]: this will work uncorrectly

File: src/modules/controllers/task.controller.js (hunk 4 above)
Comment (NataliaShchelkunova) [inline src/modules/controllers/task.controller.js]: code style

File: src/modules/controllers/task.controller.js [hunk 7]
```diff
@@ -8,28 +8,41 @@ module.exports.getAllTasks = (req, res, next) => {
...
-      res.send({ data: result });
-    });
-  }).catch(err => console.log(err));
+  const { body } = req;
+  if (body.hasOwnProperty('text')) {
+    task.save().then(() => {
+      Task.find().then(result => {
+        res.send({ data: result });
+      });
+    }).catch(err => console.log(err));
```
Comment (NataliaShchelkunova) [inline src/modules/controllers/task.controller.js]: res.status().send();

**Pull request merged**